### PR TITLE
xcp.pci: Fix xcp.pci.PCIIds.read() to support UTF-8 in hwdata/pci.ids

### PR DESCRIPTION
**Pull request by bernhardkaindl**
TL;DR:
======

- Python3 Popen() return types are version- and locale-dependent, and mode/Unicode conversion can be controlled by keyword arguments.

- For parsing text files from XAPI plugins which do not have a locale set, we need to ensure text mode with encoding="utf-8" is selected. (to read hwdata/pci.ids which has UTF-8 sequences for some graphics cards.)

Details
=======

ACK uses xcp.pci.PCIIds.read() from an XAPI plugin which does not set a locale, so it was an early example for code which is affected by Python3's Popen behavior:

Popen():
- Py2 returns Python2 strings (like bytes, we just read/write any data)
- Py3 defaults to returning bytes, not str (In Python3: str is Unicode), but it can be switched into text mode to do .decode() and .encode() for us.

  - Popen() can do encode and decode when locale encoding is set, but if not, Popen() defaults in "binary mode".
  - The locale-dependent switch is a problem because we suddenly we get bytes instead of strings on stdout and stderr.

Action:
- To guarantee text mode with UTF-8 support, we need to pass encoding="utf-8"
    - but we can only do so on Python3! (Python2 would have to ignore the encoding parameter)
- In addition, if we don't set an error handler like "replace", we'd get encoding errors if the stdout or stderr unexpectedly contains malformed UTF-8 sequences (e.g.binary data)

### CODE REVIEW

**Review by bernhardkaindl — COMMENTED** [review]
Ready for review!

**Review by psafont — APPROVED** [review]

**Comment by bernhardkaindl** [comment]
Obsoleted by #38 (merged yesterday) and #40 (ready for review)
